backtrader/feeds/influxfeed.py

```diff
@@ -0,0 +1,112 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import backtrader as bt
+import backtrader.feed as feed
+from ..utils import date2num
+from influxdb import InfluxDBClient as idbclient
+from influxdb.exceptions import InfluxDBClientError
+import datetime as dt
+
+TIMEFRAMES = dict(
+    (
+        (bt.TimeFrame.Seconds, 's'),
+        (bt.TimeFrame.Minutes, 'm'),
+        (bt.TimeFrame.Days, 'd'),
+        (bt.TimeFrame.Weeks, 'w'),
+        (bt.TimeFrame.Months, 'm'),
+        (bt.TimeFrame.Years, 'y'),
+    )
+)
+
+
+class InfluxDB(feed.DataBase):
+    params = (
+        ('host', '127.0.0.1'),
+        ('port', '8086'),
+        ('username', None),
+        ('password', None),
+        ('database', None),
+        ('timeframe', bt.TimeFrame.Days),
+        ('startdate', None),
+        ('high', 'high_p'),
+        ('low', 'low_p'),
+        ('open', 'open_p'),
+        ('close', 'close_p'),
+        ('volume', 'volume'),
+        ('ointerest', 'oi'),
+    )
+
+    def start(self):
+        super(InfluxDB, self).start()
+        try:
+            self.ndb = idbclient(self.p.host, self.p.port, self.p.username,
+                                 self.p.password, self.p.database)
+        except InfluxDBClientError as err:
+            print('Failed to establish connection to InfluxDB: %s' % err)
+
+        tf = '{multiple}{timeframe}'.format(
+            multiple=(self.p.compression if self.p.compression else 1),
+            timeframe=TIMEFRAMES.get(self.p.timeframe, 'd'))
+
+        if not self.p.startdate:
+            st = '<= now()'
+        else:
+            st = '>= \'%s\'' % self.p.startdate
+
+        # The query could already consider parameters like fromdate and todate
+        # to have the database skip them and not the internal code
+        qstr = ('SELECT mean("{open_f}") AS "open", mean("{high_f}") AS "high", '
+                'mean("{low_f}") AS "low", mean("{close_f}") AS "close", '
+                'mean("{vol_f}") AS "volume", mean("{oi_f}") AS "openinterest" '
+                'FROM "{dataname}" '
+                'WHERE time {begin} '
+                'GROUP BY time({timeframe}) fill(none)').format(
+                    open_f=self.p.open, high_f=self.p.high,
+                    low_f=self.p.low, close_f=self.p.close,
+                    vol_f=self.p.volume, oi_f=self.p.ointerest,
+                    timeframe=tf, begin=st, dataname=self.p.dataname)
+
+        try:
+            dbars = list(self.ndb.query(qstr).get_points())
```

Comment:
dbars has length of 10000
so you're limited with loading 10000 rows?

Reply:
@josefstr I seem to remember this is a limitation controlled by your influxd configuration.

Reply:
so you can load more than 10000 rows? I cannot find any limitations in the influxd configuration, could you give me a hint? Thanks :)

Reply:
there are 3 issues:
https://github.com/influxdata/influxdb/issues/8222
https://github.com/influxdata/influxdb-python/issues/450
https://github.com/influxdata/influxdb-python/issues/463

Looking at a tick time series I have more than 20 million rows per symbol, so it's not possible to load them all at once and then process.
I'm not familiar with python, but we need to iterate over the result somehow and process the incoming chunks sequentially like in a stream.

Reply:
As you point out, this appears to be an issue in influxdb-python and not backtrader.
There are other options for storing and querying time series data. You might want to look at timescale.com as an option for your larger datasets.

This should probably be discussed in the community thread.

Reply:
I'd like to open a issue on this, but they are disabled on github for this repo?

Reply:
@josefstr the issue is not with the code in this project but instead is an issue/limitation of the python module that is being used. If you can find another python module that has fixed this limitation, or convince the team responsible for the influxdb-python code to fix their issue, we should be able to implement those changes.